src/Console/Command/DispatchCommand.php

```diff
@@ -527,10 +536,74 @@ private function renderFile(Package $package, $repositoryName, $localPath, $dist
                 $stableBranch,
                 $legacyBranch,
                 $branchConfig['docs_path'],
-                str_replace(array(static::PACKAGIST_GROUP.'/', '-bundle'), '', $package->getName()),
-            ), $localContent));
+                str_replace([$this->packagistGroup . '/', '-bundle'], '', $package->getName()),
+            ], $localContent));
         }
         // Restore file permissions after content copy
         $this->fileSystem->chmod($distPath, fileperms($localPath));
     }
+
+    /**
+     * @param string $path
+     * @param array $projectConfig
+     *
+     * @throws \Exception
+     */
+    private function setVersionConstraintsInComposer($path, array $projectConfig, $currentBranch)
+    {
+        $filePath = $path.'/composer.json';
+        if (!$this->fileSystem->exists($filePath)) {
+            throw new \Exception('No composer.json found at: '.$filePath);
+        }
+        $composerAsString = file_get_contents($filePath);
+        $composerAsJson = json_decode($composerAsString, true);
+        if (!isset($composerAsJson['require'])) {
+            throw new \Exception('no require path found in composer.json at: '.$filePath);
+        }
+
+        $branchConfig = $projectConfig['branches'][$currentBranch];
+        $composerAsJson['require']['php'] = $this->evaluateVersionString($branchConfig['php']);
```

Comment:
@dbu i crate php version constraint, and the symfony version constraints in here.

Reply:
that is quite cool!